### ISSUE TITLE
Import of security_group_rule does not accept to_port 0

### ISSUE THREAD

**Issue by github-reader** [hashicorp/terraform-provider-aws#21676]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

terraform 1.0.9
provider aws 3.57.0

### Affected Resource(s)

aws_security_group_rule

* aws_XXXXX

### Terraform Configuration Files

```hcl
data "aws_security_group" "private_sg" {
  filter {
    name   = "group-name"
    values = ["private"]
  }

  filter {
    name   = "vpc-id"
    values = [var.vpc_id]
  }
}

resource "aws_security_group_rule" "ingress_1" {
  type              = "ingress"
  from_port         = 8
  to_port           = 0
  protocol          = "tcp"
  cidr_blocks       = ["<CIDR>"]
  security_group_id = data.aws_security_group.private_sg.id
  description       = "ICMP from datacenter"
```

### Debug Output

terraform-1.0.9 import aws_security_group_rule.ingress_1 <SG ID>_ingress_tcp_8_0_<CIDR>
aws_security_group_rule.ingress_1: Importing from ID "<SG ID>_ingress_tcp_8_0_<CIDR>"...
╷
│ Error: unexpected format of import string ("<SG ID>_ingress_tcp_8_0_<CIDR>"), expected SECURITYGROUPID_TYPE_PROTOCOL_FROMPORT_TOPORT_SOURCE[_SOURCE]*: invalid port

Traced log output:
2021-11-08T14:35:15.062Z [TRACE] vertex "aws_security_group_rule.ingress_1": starting visit (*terraform.NodeAbstractResource)
2021-11-08T14:35:15.062Z [TRACE] vertex "aws_security_group_rule.ingress_1": visit complete
2021-11-08T14:35:15.062Z [TRACE] vertex "aws_security_group_rule.ingress_1 (import id \"<SG ID>_ingress_tcp_8_0_172.24.0.0/16\")": starting visit (*terraform.graphNodeImportState)
2021-11-08T14:35:15.062Z [TRACE] GRPCProvider: ImportResourceState
2021-11-08T14:35:15.062Z [TRACE] vertex "aws_security_group_rule.egress_5": starting visit (*terraform.NodeAbstractResource)
2021-11-08T14:35:15.062Z [TRACE] vertex "aws_security_group_rule.egress_5": visit complete
2021-11-08T14:35:15.062Z [TRACE] vertex "aws_security_group_rule.egress_7": starting visit (*terraform.NodeAbstractResource)
2021-11-08T14:35:15.062Z [TRACE] vertex "aws_security_group_rule.egress_7": visit complete
2021-11-08T14:35:15.063Z [INFO]  provider.terraform-provider-aws_v3.57.0_x5: 2021/11/08 14:35:15 [DEBUG] Validating import string <SG ID>_ingress_tcp_8_0_<CIDR>: timestamp=2021-11-08T14:35:15.062Z
2021-11-08T14:35:15.063Z [TRACE] vertex "aws_security_group_rule.ingress_1 (import id \"<SG ID>_ingress_tcp_8_0_<CIDR>\")": visit complete
2021-11-08T14:35:15.063Z [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/aws\"] (close)" errored, so skipping
2021-11-08T14:35:15.063Z [TRACE] dag/walk: upstream of "root" errored, so skipping
╷
│ Error: unexpected format of import string ("<SG ID>_ingress_tcp_8_0_<CIDR>"), expected SECURITYGROUPID_TYPE_PROTOCOL_FROMPORT_TOPORT_SOURCE[_SOURCE]*: invalid port
│ 
│ 
╵

### Expected Behavior

The import should run successful.

### Actual Behavior

`terraform import aws_security_group_rule.ingress_1 <SG ID>_ingress_tcp_8_0_<CIDR>`

Import command produces an error:
╷
│ Error: unexpected format of import string ("<SG ID>_ingress_tcp_8_0_<CIDR>"), expected SECURITYGROUPID_TYPE_PROTOCOL_FROMPORT_TOPORT_SOURCE[_SOURCE]*: invalid port

This error had always been reported years before.
https://guthub.com/hashicorp/terraform-provider-aws/issues/6452

### Steps to Reproduce

`terraform import aws_security_group_rule.ingress_1 <SG ID>_ingress_tcp_8_0_<CIDR>`

Reason for this behaviour was found. You MUST give port numbers in ascending order to this command.

`terraform import aws_security_group_rule.ingress_1 <SG ID>_ingress_tcp_0_8_<CIDR>`

Command should return an error message, telling the user to give ports in ascending order.

### References
This error had always been reported years before.
https://github.com/hashicorp/terraform-provider-aws/issues/6452

* #0000

**Comment by justinretzolk** [comment]
Hey @github-reader 👋 Thank you for taking the time to file this issue. I took a look over the documentation for `aws_security_group_rule`, and believe that this is a result of the ordering of `from_port` and `to_port` (emphasis is mine):

> `from_port` - (Required) **Start port** (or ICMP type number if protocol is "icmp" or "icmpv6").
> `to_port` - (Required) **End port** (or ICMP code if protocol is "icmp")

My read on this is that it would expect that `from_port` is always lower than `to_port`. It looks like you've already called this out -- in your opinion, would a documentation update be enough to make this more clear?

**Comment by github-reader** [comment]
Hello @justinretzolk 

Updating the documentation is a good solution of this problem. An error message with this information would be better. I am fine with both versions.

**Comment by wedge-jarrad** [comment]
Looks like #21163 released in v3.65.0 added an explicit error message for this.

```go
		} else if p2 < p1 {
			return nil, fmt.Errorf(errStr, importStr, "to_port lower than from_port")
```
